tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/AnonymousRecords.fs

```diff
@@ -63,6 +63,28 @@ let x () : {| A: int  |} =  {| A = 123 ; B = ""; C = 1 |}
         |> withDiagnostics [
             (Error 1, Line 2, Col 29, Line 2, Col 58, "This anonymous record has extra fields. Remove fields 'B, C'.")
         ]
+        
+    [<Fact>]
+    let ``Using the wrong anon record with single field`` () =
+        Fsx """
+let x() = ({| b = 2 |} = {| a = 2 |} )
+"""
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 1, Line 2, Col 26, Line 2, Col 37, "This anonymous record should have field 'b'.")
```

Comment:
If it's not complicated, could we do this?
```suggestion
            (Error 1, Line 2, Col 26, Line 2, Col 37, "This anonymous record should have field 'b', not 'a'.")
```

Reply:
I thought about but I decided to keep it as it is to have it consistent with the rest of anon record error messages. But Yeah should be doable

Reply:
I think it describes the error better. Because just adding field 'b' won't fix the problem. 

Also might be useful if you make a typo (easy with anon record, which are usually not in many places) and it's telling you it should have field `SomeIncrediblyLongFieldName` and you think it does have it, but instead it has `SomeIncrediblyLongFieldname`. If they're both in the message you'll figure it out sooner.